src/main/java/wheresmymoney/command/StatsCommand.java

```diff
@@ -0,0 +1,87 @@
+import wheresmymoney.Parser;
+import wheresmymoney.Ui;
+import wheresmymoney.exception.WheresMyMoneyException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class StatsCommand extends Command {
+
+    public StatsCommand(HashMap<String, String> argumentsMap) {
+        super(argumentsMap);
+    }
+
+    /**
+     * Get the list of expenses based on various filter metrics
+     *
+     * @param expenseList ExpenseList to be filtered by category, a start date and an end date
+     */
+    private ArrayList<Expense> getList(ExpenseList expenseList) throws WheresMyMoneyException {
+        String statsCategory = argumentsMap.get(Parser.ARGUMENT_CATEGORY);
+        String from = argumentsMap.get(Parser.ARGUMENT_FROM);
+        String to = argumentsMap.get(Parser.ARGUMENT_TO);
+        return expenseList.listByFilter(statsCategory, from, to);
+    }
+
+    /**
+     * Display the highest expense, lowest expense and mean price of the expense list passed to it
+     *
+     * @param filteredExpenses List of expenses to be displayed
+     * @param expenseList Main expense list to retrieve expense indices
+     */
+    private void displayStats(ArrayList<Expense> filteredExpenses, ExpenseList expenseList)
+            throws WheresMyMoneyException {
+        if (filteredExpenses.isEmpty()) {
+            Ui.displayMessage("No matching expenses were found!");
+            return;
+        }
+        Expense init = filteredExpenses.get(0);
+        float sum = 0;
+        Expense highest = init;
+        Expense lowest = init;
+        for (Expense expense : filteredExpenses) {
+            float price = expense.getPrice();
+            sum += price;
+            if (price > highest.getPrice()) {
+                highest = expense;
+            }
+            if (price < lowest.getPrice()) {
+                lowest = expense;
+            }
+        }
+        float mean = sum/filteredExpenses.size();
+        Ui.displayMessage("HIGHEST EXPENSE:");
+        displayExpense(expenseList, highest);
+        Ui.displayMessage("LOWEST EXPENSE:");
+        displayExpense(expenseList, lowest);
+        Ui.displayMessage("MEAN PRICE: " + mean);
+    }
+
+    /**
+     * Display index, category, description, price, and date added of the expense passed to it
+     *
+     * @param expenseList Main expense list to retrieve expense indices
+     * @param expense Expense to be displayed
+     */
+    private static void displayExpense(ExpenseList expenseList, Expense expense) throws WheresMyMoneyException {
+        String index = expenseList.getIndexOf(expense) + 1 + ". ";
+        String category = "CATEGORY: " + expense.getCategory();
+        String description = "DESCRIPTION: " + expense.getDescription();
+        String price = "PRICE: " + expense.getPrice();
+        String dateAdded = "DATE ADDED: " + expense.getDateAdded();
+        Ui.displayMessage(index + category + ", " + description + ", " + price + ", " + dateAdded);
+    }
```

Comment:
Duplicated code from `ListCommand` class.

Suggestion: Put `displayExpense` method in `ExpenseList` class or `Ui` class and make calls to this method where necessary.